forge-ai/src/main/java/forge/ai/ComputerUtil.java

```diff
@@ -383,7 +377,8 @@ public static Card getCardPreference(final Player ai, final Card activate, final
                     }
 
                     for (String validItem : prefValid[1].split(",")) {
-                        final CardCollection prefList = CardLists.getValidCards(typeList, validItem, activate.getController(), activate, null);
+                        CardCollection prefList = CardLists.getValidCards(typeList, validItem, activate.getController(), activate, null);
+                        prefList = CardLists.filter(prefList, card -> !card.hasSVar("AIDontSacrifice"));
```

Comment:
Not sure this new SVar makes sense?
I mean the P9 are that mostly for their cheap costs, but if you already played them they should be ok to sac (compared to other artifacts anyway)?

Reply:
The AI may be too eager to sacrifice them at the first available opportunity otherwise... Like, let's say it plays a turn 1 land and Mox, then on turn 2, it plays something and sacs the Mox even though it could have kept it for the early game mana advantage (having 3 mana sources on turn 2 instead of having 2 mana sources), something like that. Think it makes sense? Or doesn't matter much?

Reply:
If it makes sense to keep this at all, maybe we can change it so that the AI sacrifices them if it has enough mana sources or something like that, what do you think?

Reply:
right, I just don't like extra checks like that if they only improve one scenario type while worsening others

Reply:
from a logic point, it would make sense to have some kind of counterpart to SacCost which instead of such source can be put on potential victims?
maybe some logic from NeedsToPlay can be reused

Reply:
Yeah, I agree with the reasoning, I guess let's keep this part out of this particular MR, I need to think a bit more on how to best implement this... I'll keep the logic in the "default auto-sac preference" part for the time being, as it's a bit too risky to leave with the Moxen freely sacrifice-able :)